[SecuritySolution] Disable the timeline save tour with localstorage flag

janmonschke opened the pull request:
## Summary

In this PR we're getting rid of the `enableExperimental` flag `disableTimelineSaveTour` in favor of a custom localStorage flag in the cypress tests. This will allow us to run the tests against a production version of Kibana without setting custom config flags.

janmonschke committed:
disable the timeline save tour with localstorage flag instead of `enableExperimental`

MadameSheema reviewed (approved):
security-engineering-productivity changes LGTM, thanks!

janmonschke commented:
@elasticmachine merge upstream

janmonschke commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into securitySolution/timelines/disable-timeline-save-tour-in-localstorage

janmonschke committed:
a more defensive approach to reading from LS

Also changes the key, which will re-show the "tour" to people in serverless that have already seen it

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

kpollich reviewed (approved):
Fleet changes

janmonschke commented on x-pack/plugins/security_solution/public/timelines/components/timeline/header/save_timeline_button.tsx:
```diff
@@ -56,27 +53,28 @@ export const SaveTimelineButton = React.memo<SaveTimelineButtonProps>(({ timelin
+  const [timelineTourStatus, setTimelineTourStatus] = useLocalStorage({
+    defaultValue: { isTourActive: true },
     key: LOCAL_STORAGE_KEY,
+    isInvalidDefault: (valueFromStorage) => {
```
@logeekal The key in that file is prefixed with `securitySolution` and this one isn't. The prefix is added by `useLocalStorage`. So sadly we have to duplicate it here...

janmonschke replied:
@logeekal Sure, that would be ideal but this piece of code makes this pretty hard without changing the hook in this PR:

https://github.com/elastic/kibana/blob/b55dae32f67a5cd2fb106c57a9712f443e7492f2/x-pack/plugins/security_solution/public/common/components/local_storage/index.tsx#L33

logeekal replied:
Understood. in that case I think it is okay to go ahead. I am, anyway, creating a new timeline tour already, which will override the current tour. 

But since that is multistep tour, I am thinking of using same strategy `Guided Rule Tour` takes as in below file by plugging in guidedOnboardingApi:

https://github.com/elastic/kibana/blob/b55dae32f67a5cd2fb106c57a9712f443e7492f2/x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/alternative_tour/tour.tsx#L58-L60

logeekal reviewed (commented):
Looks good. 🚀
